tests/helpers/test_entityfilter.py

```diff
@@ -78,6 +137,27 @@ def test_exclude_domain_case4b():
     assert testfilter("sun.sun") is True
 
 
+def test_exclude_glob_case4b():
+    """Test case 4b - include and exclude specified, with excluded glob."""
+    incl_dom = {}
+    incl_glob = {}
+    incl_ent = {"binary_sensor.working"}
+    excl_dom = {}
+    excl_glob = {"binary_sensor.*"}
+    excl_ent = {"light.ignoreme", "sensor.notworking"}
+    testfilter = generate_filter(
+        incl_dom, incl_ent, excl_dom, excl_ent, incl_glob, excl_glob
+    )
+
+    assert testfilter("sensor.test")
+    assert testfilter("sensor.notworking") is False
+    assert testfilter("light.test")
+    assert testfilter("light.ignoreme") is False
+    assert testfilter("binary_sensor.working")
+    assert testfilter("binary_sensor.another") is False
+    assert testfilter("sun.sun") is True
+
+
 def test_no_domain_case4c():
```

Comment:
We should not alter the behavior of existing tests beyond that is needed to make the function not reject argument changes, it is better to add new ones so we prove the old tests are still working without changes.